src/Compilers/CSharp/Portable/Binder/Semantics/Operators/BinaryOperatorOverloadResolution.cs

```diff
@@ -1311,19 +1312,16 @@ private BetterResult MoreSpecificOperator(BinaryOperatorSignature op1, BinaryOpe
                 op2Right = op2.RightType;
             }
 
-            var uninst1 = ArrayBuilder<TypeSymbol>.GetInstance();
-            var uninst2 = ArrayBuilder<TypeSymbol>.GetInstance();
+            using var uninst1 = TemporaryArray<TypeSymbol>.Empty;
+            using var uninst2 = TemporaryArray<TypeSymbol>.Empty;
```

Comment:
📝 `TemporaryArray<T>` stores up to 4 elements on the stack, and only uses `ArrayBuilder<T>.GetInstance()` for 5 or more elements.

Reply:
Being pinned to a core might make sense as well.

Reply:
There are options there, but it's difficult to implement them without penalizing some other scenario. Stack-based allocations have done a good job of addressing this situation in IDE code, so I would expect them to do the same here.